src/Symfony/Component/Config/Resource/ComposerResource.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Resource;
+
+/**
+ * ComposerResource tracks the PHP version and composer dependencies.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class ComposerResource implements SelfCheckingResourceInterface, \Serializable
+{
+    private $versions;
+    private $vendors;
+
+    private static $globalVersions;
+    private static $globalVendors;
+
+    public function __construct()
+    {
+        self::refresh();
+        $this->versions = self::$globalVersions;
+        $this->vendors = self::$globalVendors;
+    }
+
+    public function getVendors()
+    {
+        return array_keys($this->vendors);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __toString()
+    {
+        return __CLASS__;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isFresh($timestamp)
+    {
+        self::refresh();
+
+        if (self::$globalVersions !== $this->versions) {
+            return false;
+        }
+
+        return self::$globalVendors === $this->vendors;
+    }
+
+    public function serialize()
+    {
+        return serialize(array($this->versions, $this->vendors));
+    }
+
+    public function unserialize($serialized)
+    {
+        list($this->versions, $this->vendors) = unserialize($serialized);
+    }
+
+    private static function refresh()
+    {
+        if (null !== self::$globalVersions) {
+            return;
+        }
+
+        self::$globalVersions = array();
+
+        foreach (get_loaded_extensions() as $ext) {
+            self::$globalVersions[$ext] = phpversion($ext);
+        }
+
+        self::$globalVendors = array();
+
+        foreach (get_declared_classes() as $class) {
+            if ('C' === $class[0] && 0 === strpos($class, 'ComposerAutoloaderInit')) {
+                $r = new \ReflectionClass($class);
+                $v = dirname(dirname($r->getFileName())).DIRECTORY_SEPARATOR;
+                if (file_exists($v.'composer/installed.json')) {
+                    self::$globalVendors[$v] = @filemtime($v);
```

Comment:
IMO, we should rather hash the file. If you rerun ``composer install``, the file will be touched by composer even when there is no change